core/src/toga/paths.py

```diff
@@ -40,13 +40,15 @@ def config(self) -> Path:
         """The platform-appropriate location for storing user configuration
         files associated with this app.
         """
+        self._impl.get_config_path().mkdir(parents=True, exist_ok=True)
         return self._impl.get_config_path()
```

Comment:
This does exactly what is required, but it makes 2 calls on the underlying `_impl` API. The value won't change between those two calls, so we can store and re-use that value.
```suggestion
        path = self._impl.get_config_path()
        path.mkdir(parents=True, exist_ok=True)
        return path
```

The same optimisation will exist for the other attributes as well.